activemodel/lib/active_model/attribute.rb

```diff
@@ -96,6 +96,14 @@ def with_type(type)
       end
     end
 
+    def deep_dup # :nodoc:
```

Comment:
Could we apply this optimization on `dup` in general? The caller is nearby, so it's hardly a layering violation, but it still feels just a little more implementation-aware than we could be, maybe?

(Incidentally notable: our actual `initialize_dup`'s value-dupping seems surprisingly casual -- likely doing so when unnecessary, possibly missing necessary depth in other cases?)

Reply:
> The caller is nearby, so it's hardly a layering violation, but it still feels just a little more implementation-aware than we could be, maybe?

Yeah, maybe we should just not call `dup` nor `deep_dup` but a more explicit method? something like `deep_dup_unless_shareable` (or something better that convey the same meaning)

Reply:
Look like that broke CI quite a bit 🤔

Reply:
Nevermind, the issue was that I changed `attributes` for `@attributes`, some subclass do redefine that accessor.